azurerm/internal/services/network/resource_arm_express_route_gateway.go

```diff
@@ -0,0 +1,203 @@
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		resp, err := client.Get(ctx, resourceGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Error checking for present of existing ExpressRoute Gateway %q (Resource Group %q): %+v", name, resourceGroup, err)
+			}
+		}
+		if resp.ID != nil && *resp.ID != "" {
+			return tf.ImportAsExistsError("azurerm_express_route_gateway", *resp.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	virtualHubId := d.Get("virtual_hub_id").(string)
+	t := d.Get("tags").(map[string]interface{})
+
+	parameters := network.ExpressRouteGateway{
+		Location: utils.String(location),
+		ExpressRouteGatewayProperties: &network.ExpressRouteGatewayProperties{
+			AutoScaleConfiguration: &network.ExpressRouteGatewayPropertiesAutoScaleConfiguration{
+				Bounds: expandExpressRouteGatewayAutoScaleConfigurationBounds(d),
+			},
+			VirtualHub: &network.VirtualHubID{
+				ID: &virtualHubId,
+			},
+		},
+		Tags: tags.Expand(t),
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, name, parameters)
+	if err != nil {
+		return fmt.Errorf("Error creating ExpressRoute Gateway %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for creation of ExpressRoute Gateway %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving ExpressRoute Gateway %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if resp.ID == nil || *resp.ID == "" {
+		return fmt.Errorf("Cannot read ExpressRoute Gateway %q (Resource Group %q) ID", name, resourceGroup)
+	}
+	d.SetId(*resp.ID)
+
+	return resourceArmExpressRouteGatewayRead(d, meta)
+}
+
+func resourceArmExpressRouteGatewayRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.ExpressRouteGatewaysClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["expressRouteGateways"]
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] ExpressRoute Gateway %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading ExpressRoute Gateway %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	if props := resp.ExpressRouteGatewayProperties; props != nil {
+		d.Set("virtual_hub_id", props.VirtualHub.ID)
```

Comment:
there's a crash here if VirtualHub is nil (which can happen with an invalid API response since Azure returns the representation of the resource at the last time it was modified) - as such can we make this:

```suggestion
		virtualHubId := ""
		if props.VirtualHub != nil && props.VirtualHub.ID != nil {
			virtualHubId = *props.VirtualHub.ID
		}
		d.Set("virtual_hub_id", virtualHubId)
```